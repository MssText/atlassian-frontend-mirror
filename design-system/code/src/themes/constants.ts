export const CODE_FONT_SIZE = 12;
export const CODE_LINE_HEIGHT = '20px';

// css vars used to decrease font family being repeatedly declared inline
export const VAR_CODE_FONT_FAMILY = '--code-font-family';
export const VAR_CODE_FONT_FAMILY_ITALIC = `${VAR_CODE_FONT_FAMILY}-italic`;
export const VAR_CODE_LINE_NUMBER_BG_COLOR = '--line-number-bg-color';
