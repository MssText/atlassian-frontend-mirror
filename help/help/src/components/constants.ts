export const MIN_CHARACTERS_FOR_SEARCH = 3;
export const LOADING_TIMEOUT = 1000;

export enum VIEW {
  DEFAULT_CONTENT = 'DEFAULT_CONTENT',
  ARTICLE = 'ARTICLE',
  ARTICLE_NAVIGATION = 'ARTICLE_NAVIGATION',
  SEARCH = 'SEARCH',
}

// Animation related consts
export const TRANSITION_DURATION_MS = 300;
export const SEARCH_RESULTS_FADEIN_TRANSITION_DURATION_MS = 440;

export enum TRANSITION_STATUS {
  UNMOUNTED = 'unmounted',
  EXITED = 'exited',
  ENTERING = 'entering',
  ENTERED = 'entered',
  EXITING = 'exiting',
}
