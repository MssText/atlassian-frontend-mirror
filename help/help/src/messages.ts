import { defineMessages } from 'react-intl';

export const messages = defineMessages({
  help_loading: {
    id: 'help.loading',
    defaultMessage: 'Loading',
    description: '',
  },
  help_header: {
    id: 'help.header',
    defaultMessage: 'Help',
    description: '',
  },
  help_close: {
    id: 'help.close',
    defaultMessage: 'Close',
    description: '',
  },
  help_navigation_back: {
    id: 'help.navigation.back',
    defaultMessage: 'Back',
    description: '',
  },
  help_search_placeholder: {
    id: 'help.search.placeholder',
    defaultMessage: 'Search help articles',
    description: '',
  },
  help_article_rating_title: {
    id: 'help.article_rating.title',
    defaultMessage: 'Was this helpful?',
    description: '',
  },
  help_article_rating_option_yes: {
    id: 'help.article_rating.option.yes',
    defaultMessage: 'Yes',
    description: '',
  },
  help_article_rating_option_no: {
    id: 'help.article_rating.option.no',
    defaultMessage: 'No',
    description: '',
  },
  help_article_rating_form_title: {
    id: 'help.article_rating.form.title',
    defaultMessage: 'Tell us why you gave this rating',
    description: '',
  },
  help_article_rating_accurate: {
    id: 'help.article_rating.accurate',
    defaultMessage: `It isn't accurate`,
    description: '',
  },
  help_article_rating_clear: {
    id: 'help.article_rating.clear',
    defaultMessage: `It isn't clear`,
    description: '',
  },
  help_article_rating_relevant: {
    id: 'help.article_rating.relevant',
    defaultMessage: `It isn't relevant`,
    description: '',
  },
  help_article_rating_form_submit: {
    id: 'help.article_rating.form.submit',
    defaultMessage: `Submit`,
    description: '',
  },
  help_article_rating_form_Success: {
    id: 'help.article_rating.form.success',
    defaultMessage: `Thanks!`,
    description: '',
  },
  help_article_rating_form_failed: {
    id: 'help.article_rating.form.failed',
    defaultMessage: `We couldn't submit your feedback.`,
    description: '',
  },
  help_article_rating_form_failed_try_again: {
    id: 'help.article_rating.form.failed.try_again',
    defaultMessage: `Try again`,
    description: '',
  },
  help_article_rating_form_cancel: {
    id: 'help.article_rating.form.cancel',
    defaultMessage: `Cancel`,
    description: '',
  },
  help_article_rating_form_contact_me: {
    id: 'help.article_rating.form.contact_me',
    defaultMessage: `Atlassian can contact me about this feedback`,
    description: '',
  },
  help_article_list_item_type_whats_new: {
    id: 'help.article_list_item.type.whats_new',
    defaultMessage: `WHAT'S NEW`,
    description: 'Article type title',
  },
  help_article_list_item_type_help_article: {
    id: 'help.article_list_item.type.help-article',
    defaultMessage: `HELP ARTICLE`,
    description: 'Article type title',
  },
  help_article_list_show_more: {
    id: 'help.article_list.show_more',
    defaultMessage: 'Show {numberOfRelatedArticlesLeft} more articles',
    description: '',
  },
  help_article_list_show_less: {
    id: 'help.article_list.show_less',
    defaultMessage: 'Show less',
    description: '',
  },
  help_related_article_title: {
    id: 'help.related_article.title',
    defaultMessage: 'Related articles',
    description: 'Title of the related articles list',
  },
  help_related_article_endpoint_error_title: {
    id: 'help.related_article.endpoint-error.title',
    defaultMessage: `We're having some trouble`,
    description: '',
  },
  help_related_article_endpoint_error_description: {
    id: 'help.related_article.endpoint-error.description',
    defaultMessage: `It's taking us longer than expected to show this content. It's provably a temporary problem.`,
    description: '',
  },
  help_related_article_endpoint_error_button_label: {
    id: 'help.related_article.endpoint-error.button-label',
    defaultMessage: `Try Again`,
    description: '',
  },
  help_search_results_search_external_site: {
    id: 'help.search_results.search_external_site',
    defaultMessage: `Can't find what you're looking for? Try again with a different term or `,
  },
  help_search_results_no_results: {
    id: 'help.search_results.no_results',
    defaultMessage: `We can't find anything matching your search`,
  },
  help_search_results_no_results_line_two: {
    id: 'help.search_results.no_results_line_two',
    defaultMessage: `Try again with a different term or `,
  },
  help_search_results_external_site_link: {
    id: 'help.search_results.external_site_link',
    defaultMessage: `search all online help articles.`,
  },
  help_article_error_title: {
    id: 'help.article_error.title',
    defaultMessage: `Something's gone wrong`,
    description: '',
  },
  help_article_error_text: {
    id: 'help.article_error.text',
    defaultMessage: `Something prevented the page from loading. We will try to reconect automatically`,
    description: '',
  },
  help_article_error_button_label: {
    id: 'help.article_error.button_label',
    defaultMessage: `Try now`,
    description: '',
  },
  help_search_error: {
    id: 'help.search_error',
    defaultMessage: `We are having some trouble`,
    description: '',
  },
  help_search_error_line_two: {
    id: 'help.search_error_line_two',
    defaultMessage: `It's taking us longer than expected to show this content. It's probably a temporary problem.`,
    description: '',
  },
  help_search_error_button_label: {
    id: 'help.search_error.button_label',
    defaultMessage: `Retry`,
    description: '',
  },
});
