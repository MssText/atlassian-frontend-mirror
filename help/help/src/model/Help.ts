import { Article, ArticleItem, ArticleFeedback } from './Article';
import { REQUEST_STATE } from './Requests';
import { UIAnalyticsEvent } from '@atlaskit/analytics-next';

export interface HistoryItem {
  uid: number;
  id: string;
  state: REQUEST_STATE;
  article?: Article;
}

export interface Help {
  // Id of the article to display. This prop is optional, if is not defined the default content will be displayed
  articleId?: string;
  // Setter for the articleId. This prop is optional, if is not defined, the back button will not be visible
  articleIdSetter?(id: string): void;
  // Function used to get an article content. This prop is optional, if is not defined the default content will be displayed
  onGetArticle?(id: string): Promise<Article>;
  // History. This prop is optional
  history?: HistoryItem[];
  // Setter for the history. This prop is optional
  historySetter?(history: HistoryItem[]): void;
  // Event handler fired when the user clicks the "Try Again" button (in the article loading fail screen)
  onArticleLoadingFailTryAgainButtonClick?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
    articleId: string,
  ): void;
  // Function used to get related articles. This prop is optional, if is not defined the related articles will not be displayed
  onGetRelatedArticleOfOpenArticle?(
    routeGroup?: string,
    routeName?: string,
  ): Promise<ArticleItem[]>;
  // Function executed when the user click a related article item
  onRelatedArticlesListItemClick?: (
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
    articleData: ArticleItem,
  ) => void;
  // Function used to search an article.  This prop is optional, if is not defined search input will be hidden
  onSearch?(value: string): Promise<ArticleItem[]>;
  // Event handler fired when the search input changes
  onSearchInputChanged?(
    event: React.KeyboardEvent<HTMLInputElement>,
    analyticsEvent: UIAnalyticsEvent,
    value: string,
  ): void;
  // Event handler fired when the search input changes
  onSearchInputCleared?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
  ): void;
  // Event handler fired when the user clicks an Item in the search result list. This prop is optional
  onSearchResultItemClick?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
    articleData: ArticleItem,
  ): void;
  // External search site URL
  searchExternalUrl?: string;
  // Event handler for the External search site Click. This prop is optional
  onSearchExternalUrlClick?(
    event?: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent?: UIAnalyticsEvent,
  ): void;
  // Event handler for the close button. This prop is optional, if this function is not defined the close button will not be displayed
  onCloseButtonClick?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
  ): void;
  // Event handler for the "Was this helpful" submit button. This prop is optional, if is not defined the "Was this helpful" section will be hidden
  onWasHelpfulSubmit?(
    analyticsEvent: UIAnalyticsEvent,
    articleFeedback: ArticleFeedback,
    articleData: ArticleItem,
  ): Promise<boolean>;
  // Event handler for the "Yes" button of the "Was this helpful" section. This prop is optional
  onWasHelpfulYesButtonClick?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
    ArticleItem: ArticleItem,
  ): void;
  // Event handler for the "No" button of the "Was this helpful" section. This prop is optional
  onWasHelpfulNoButtonClick?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
    ArticleItem: ArticleItem,
  ): void;
  // Event handler for the "Back" button. This prop is optional
  onBackButtonClick?(
    event: React.MouseEvent<HTMLElement, MouseEvent>,
    analyticsEvent: UIAnalyticsEvent,
  ): void;
  // Function executed when the article rendering begins
  onArticleRenderBegin?(): void;
  // Function executed when the article rendering finishes
  onArticleRenderDone?(): void;
  // Footer content. This prop is optional
  footer?: React.ReactNode;
  // Wrapped content
  children?: React.ReactNode;
}
