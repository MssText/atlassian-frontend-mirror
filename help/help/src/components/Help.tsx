import React from 'react';
import {
  withAnalyticsEvents,
  WithAnalyticsEventsProps,
  withAnalyticsContext,
} from '@atlaskit/analytics-next';
import { defaultAnalyticsAttributes } from '../analytics';
import { Help as HelpInterface } from '../model/Help';

import { HelpContextProvider } from './HelpContext';
import MessagesIntlProvider from './MessagesIntlProvider';

import HelpContent from './HelpContent';

export type Props = HelpInterface & WithAnalyticsEventsProps;

export class Help extends React.PureComponent<Props> {
  render() {
    const { children, ...rest } = this.props;
    return (
      <HelpContextProvider {...rest} defaultContent={children}>
        <MessagesIntlProvider>
          <HelpContent />
        </MessagesIntlProvider>
      </HelpContextProvider>
    );
  }
}

export default withAnalyticsContext(defaultAnalyticsAttributes)(
  withAnalyticsEvents()(Help),
);
