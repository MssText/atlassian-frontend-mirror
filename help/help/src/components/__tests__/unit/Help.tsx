import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render, wait, fireEvent, act } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import {
  getMockArticle,
  getMockArticleItemList,
} from '../../../util/testing/mock';
import { messages } from '../../../messages';

import { TRANSITION_DURATION_MS } from '../../constants';
import { Help } from '../../Help';

// Messages
const intlProvider = new IntlProvider({ locale: 'en' });
const { intl } = intlProvider.getChildContext();
const messageClose = intl.formatMessage(messages.help_close);
const messageBack = intl.formatMessage(messages.help_navigation_back);

// Mock props
let articleId = '';
const mockArticleIdSetter = jest
  .fn()
  .mockImplementation((id) => (articleId = id));
const mockOnGetArticle = jest.fn().mockImplementation(
  (id) =>
    new Promise((resolve, reject) => {
      if (id > 0) {
        setTimeout(() => {
          resolve(getMockArticle(id));
        }, 100);
      } else {
        reject('error');
      }
    }),
);
const mockOnArticleLoadingFailTryAgainButtonClick = jest.fn();
const mockOnGetRelatedArticleOfOpenArticle = jest.fn().mockImplementation(
  () =>
    new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve(getMockArticleItemList(10));
      }, 100);
    }),
);
const mockOnRelatedArticlesListItemClick = jest.fn();
const mockOnSearch = jest.fn().mockImplementation(
  () =>
    new Promise((resolve, reject) => {
      setTimeout(() => {
        resolve(getMockArticleItemList(10));
      }, 100);
    }),
);
const mockOnSearchInputChanged = jest.fn();
const mockOnSearchInputCleared = jest.fn();
const mockOnSearchResultItemClick = jest.fn();
const mockSearchExternalUrl = 'https://www.atlassian.com/';
const mockOnSearchExternalUrlClick = jest.fn();
const mockOnCloseButtonClick = jest.fn();
const mockOnWasHelpfulYesButtonClick = jest.fn();
const mockOnWasHelpfulNoButtonClick = jest.fn();
const mockOnBackButtonClick = jest.fn();

const defaultContentText = 'Default content';

describe('BackButton', () => {
  // jest.useFakeTimers();

  beforeEach(() => {
    //   jest.clearAllTimers();
    articleId = '';
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should match snapshot', () => {
    const { container } = render(
      <IntlProvider locale="en">
        <Help
          articleId={articleId}
          articleIdSetter={mockArticleIdSetter}
          onGetArticle={mockOnGetArticle}
          onArticleLoadingFailTryAgainButtonClick={
            mockOnArticleLoadingFailTryAgainButtonClick
          }
          onGetRelatedArticleOfOpenArticle={
            mockOnGetRelatedArticleOfOpenArticle
          }
          onRelatedArticlesListItemClick={mockOnRelatedArticlesListItemClick}
          onSearch={mockOnSearch}
          onSearchInputChanged={mockOnSearchInputChanged}
          onSearchInputCleared={mockOnSearchInputCleared}
          onSearchResultItemClick={mockOnSearchResultItemClick}
          searchExternalUrl={mockSearchExternalUrl}
          onSearchExternalUrlClick={mockOnSearchExternalUrlClick}
          onCloseButtonClick={mockOnCloseButtonClick}
          onWasHelpfulYesButtonClick={mockOnWasHelpfulYesButtonClick}
          onWasHelpfulNoButtonClick={mockOnWasHelpfulNoButtonClick}
          onBackButtonClick={mockOnBackButtonClick}
        >
          <div>{defaultContentText}</div>
        </Help>
      </IntlProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should display the close button if the props mockOnCloseButtonClick is defined', () => {
    const { queryByLabelText } = render(
      <IntlProvider locale="en">
        <Help
          articleId={articleId}
          articleIdSetter={mockArticleIdSetter}
          onGetArticle={mockOnGetArticle}
          onArticleLoadingFailTryAgainButtonClick={
            mockOnArticleLoadingFailTryAgainButtonClick
          }
          onGetRelatedArticleOfOpenArticle={
            mockOnGetRelatedArticleOfOpenArticle
          }
          onRelatedArticlesListItemClick={mockOnRelatedArticlesListItemClick}
          onSearch={mockOnSearch}
          onSearchInputChanged={mockOnSearchInputChanged}
          onSearchInputCleared={mockOnSearchInputCleared}
          onSearchResultItemClick={mockOnSearchResultItemClick}
          searchExternalUrl={mockSearchExternalUrl}
          onCloseButtonClick={mockOnCloseButtonClick}
          onSearchExternalUrlClick={mockOnSearchExternalUrlClick}
          onWasHelpfulYesButtonClick={mockOnWasHelpfulYesButtonClick}
          onWasHelpfulNoButtonClick={mockOnWasHelpfulNoButtonClick}
          onBackButtonClick={mockOnBackButtonClick}
        >
          <div>{defaultContentText}</div>
        </Help>
      </IntlProvider>,
    );

    const closeButton = queryByLabelText(messageClose);

    expect(closeButton).not.toBeNull();
  });

  it('Should NOT display the close button if the props mockOnCloseButtonClick is undefined or null', () => {
    const { queryByLabelText } = render(
      <IntlProvider locale="en">
        <Help
          articleId={articleId}
          articleIdSetter={mockArticleIdSetter}
          onGetArticle={mockOnGetArticle}
          onArticleLoadingFailTryAgainButtonClick={
            mockOnArticleLoadingFailTryAgainButtonClick
          }
          onGetRelatedArticleOfOpenArticle={
            mockOnGetRelatedArticleOfOpenArticle
          }
          onRelatedArticlesListItemClick={mockOnRelatedArticlesListItemClick}
          onSearch={mockOnSearch}
          onSearchInputChanged={mockOnSearchInputChanged}
          onSearchInputCleared={mockOnSearchInputCleared}
          onSearchResultItemClick={mockOnSearchResultItemClick}
          searchExternalUrl={mockSearchExternalUrl}
          onSearchExternalUrlClick={mockOnSearchExternalUrlClick}
          onWasHelpfulYesButtonClick={mockOnWasHelpfulYesButtonClick}
          onWasHelpfulNoButtonClick={mockOnWasHelpfulNoButtonClick}
          onBackButtonClick={mockOnBackButtonClick}
        >
          <div>{defaultContentText}</div>
        </Help>
      </IntlProvider>,
    );

    const closeButton = queryByLabelText(messageClose);

    expect(closeButton).toBeNull();
  });

  it('Should load and display an article if the articleId != ""', async () => {
    jest.useFakeTimers();
    const { queryByText } = render(
      <IntlProvider locale="en">
        <Help
          articleId={'1'}
          articleIdSetter={mockArticleIdSetter}
          onGetArticle={mockOnGetArticle}
          onArticleLoadingFailTryAgainButtonClick={
            mockOnArticleLoadingFailTryAgainButtonClick
          }
          onSearch={mockOnSearch}
          onSearchInputChanged={mockOnSearchInputChanged}
          onSearchInputCleared={mockOnSearchInputCleared}
          onSearchResultItemClick={mockOnSearchResultItemClick}
          searchExternalUrl={mockSearchExternalUrl}
          onSearchExternalUrlClick={mockOnSearchExternalUrlClick}
          onWasHelpfulYesButtonClick={mockOnWasHelpfulYesButtonClick}
          onWasHelpfulNoButtonClick={mockOnWasHelpfulNoButtonClick}
          onBackButtonClick={mockOnBackButtonClick}
        >
          <div>{defaultContentText}</div>
        </Help>
      </IntlProvider>,
    );

    // onGetArticle promise gets resolved after 100ms
    act(() => jest.advanceTimersByTime(100));

    const defaultContentElm = queryByText(defaultContentText);
    expect(defaultContentElm).not.toBeNull();

    // Once the article gets loaded
    if (defaultContentElm) {
      const defaultContentContainerElm = defaultContentElm.parentElement;

      // the defualt content should be visible until the article fade-in animation finishes
      expect(window.getComputedStyle(defaultContentContainerElm!).display).toBe(
        'block',
      );
      act(() => jest.advanceTimersByTime(TRANSITION_DURATION_MS));
      expect(window.getComputedStyle(defaultContentContainerElm!).display).toBe(
        'none',
      );
    }

    jest.clearAllTimers();
  });

  it('Should display the Back button if an article is open', async () => {
    jest.useFakeTimers();

    const { queryByText } = render(
      <IntlProvider locale="en">
        <Help
          articleId={'1'}
          articleIdSetter={mockArticleIdSetter}
          onGetArticle={mockOnGetArticle}
          onArticleLoadingFailTryAgainButtonClick={
            mockOnArticleLoadingFailTryAgainButtonClick
          }
          onSearch={mockOnSearch}
          onSearchInputChanged={mockOnSearchInputChanged}
          onSearchInputCleared={mockOnSearchInputCleared}
          onSearchResultItemClick={mockOnSearchResultItemClick}
          searchExternalUrl={mockSearchExternalUrl}
          onSearchExternalUrlClick={mockOnSearchExternalUrlClick}
          onWasHelpfulYesButtonClick={mockOnWasHelpfulYesButtonClick}
          onWasHelpfulNoButtonClick={mockOnWasHelpfulNoButtonClick}
          onBackButtonClick={mockOnBackButtonClick}
        >
          <div>{defaultContentText}</div>
        </Help>
      </IntlProvider>,
    );

    // onGetArticle promise gets resolved after 100ms
    act(() => jest.advanceTimersByTime(100));

    // Wait until the fade-in transition of the back button finishes
    act(() => jest.advanceTimersByTime(TRANSITION_DURATION_MS));
    const BackButton = queryByText(messageBack)!.closest('button');
    expect(BackButton).not.toBeNull();

    // Once the article gets loaded
    if (BackButton) {
      fireEvent.click(BackButton);

      // Wait until the fade-out transition of the back button finishes
      act(() => jest.advanceTimersByTime(TRANSITION_DURATION_MS));

      await wait(() => expect(mockOnBackButtonClick).toHaveBeenCalledTimes(1));
    }

    jest.clearAllTimers();
  });
});
