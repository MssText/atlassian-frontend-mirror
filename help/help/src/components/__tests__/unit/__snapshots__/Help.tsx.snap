// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Should match snapshot 1`] = `
.emotion-16 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #ffffff;
}

.emotion-15 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 0;
}

.emotion-4 {
  height: 56px;
  background-color: #FAFBFC;
  border-bottom: 2px solid #EBECF0;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.emotion-0 {
  color: #42526E;
  text-align: center;
  font-size: 1rem;
  font-weight: 600;
  line-height: 56px;
  width: 100%;
  display: inline-block;
  vertical-align: middle;
}

.emotion-3 {
  position: absolute;
  right: 8px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.emotion-2,
.emotion-2:hover,
.emotion-2:active,
.emotion-2:focus,
.emotion-2:visited,
.emotion-2:disabled,
.emotion-2[disabled] {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-width: 0;
  border-radius: 3px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: inherit;
  font-style: normal;
  font-family: inherit;
  font-weight: 500;
  max-width: 100%;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  transition: background 0.1s ease-out,box-shadow 0.15s cubic-bezier(0.47,0.03,0.49,1.38);
  white-space: nowrap;
  background: none;
  color: #42526E !important;
  cursor: pointer;
  height: 2.2857142857142856em;
  line-height: 2.2857142857142856em;
  padding: 0 2px;
  vertical-align: middle;
  width: auto;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
}

.emotion-2::-moz-focus-inner,
.emotion-2:hover::-moz-focus-inner,
.emotion-2:active::-moz-focus-inner,
.emotion-2:focus::-moz-focus-inner,
.emotion-2:visited::-moz-focus-inner,
.emotion-2:disabled::-moz-focus-inner,
.emotion-2[disabled]::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-1 {
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  opacity: 1;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 0;
  font-size: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin: 0 2px;
}

.emotion-14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-height: 0;
  position: relative;
  overflow-y: auto;
  overflow-x: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-9 {
  padding: 24px 16px 8px 16px;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  width: 100%;
  box-sizing: border-box;
}

.emotion-8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FAFBFC;
  border-color: #DFE1E6;
  border-radius: 3px;
  border-width: 2px;
  border-style: solid;
  box-sizing: border-box;
  color: #091E42;
  cursor: text;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  font-size: 14px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 100%;
  overflow: hidden;
  -webkit-transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  word-wrap: break-word;
  vertical-align: top;
  pointer-events: auto;
}

.emotion-5 {
  width: 24px;
  height: 24px;
  padding-left: 4px;
}

.emotion-5 > span {
  padding-left: 4px;
  height: 24px;
  width: 24px;
  box-sizing: border-box;
}

.emotion-6 {
  background-color: transparent;
  border: 0;
  box-sizing: border-box;
  color: inherit;
  cursor: inherit;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  min-width: 0;
  outline: none;
  padding: 8px 6px;
  width: 100%;
  height: 2.5714285714285716em;
  line-height: 1.4285714285714286;
}

.emotion-6[disabled] {
  -webkit-text-fill-color: unset;
  -webkit-opacity: 1px;
}

.emotion-6::-ms-clear {
  display: none;
}

.emotion-6:invalid {
  box-shadow: none;
}

.emotion-6::-webkit-input-placeholder {
  color: #7A869A;
}

.emotion-6::-moz-placeholder {
  color: #7A869A;
}

.emotion-6:-ms-input-placeholder {
  color: #7A869A;
}

.emotion-6::placeholder {
  color: #7A869A;
}

.emotion-7 {
  font-size: 0;
  padding-right: 8px;
  position: relative;
  white-space: nowrap;
}

.emotion-7 > span {
  padding-right: 4px;
}

.emotion-7 > button,
.emotion-7 > span {
  display: inline-block;
  vertical-align: middle;
}

.emotion-13 {
  width: 100%;
  box-sizing: border-box;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
  position: relative;
  overflow-x: hidden;
  overflow-y: scroll;
}

.emotion-10 {
  position: absolute;
  height: 100%;
  width: 100%;
  top: 0;
  background-color: #ffffff;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  overflow-x: hidden;
  overflow-y: auto;
  z-index: 10;
  padding: 0 16px;
}

.emotion-11 {
  padding: 16px 24px;
  position: absolute;
  height: 100%;
  width: 100%;
  top: 0;
  background-color: #ffffff;
  left: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  overflow-x: hidden;
  overflow-y: auto;
  z-index: 1;
}

.emotion-12 {
  display: block;
  height: 100%;
  overflow: auto;
}

<div
  class="emotion-16"
>
  <div
    class="emotion-15"
  >
    <div
      class="emotion-4"
    >
      <div
        class="emotion-0"
      >
        Help
      </div>
      <div
        class="emotion-3"
      >
        <div>
          <button
            class="emotion-2"
            tabindex="0"
            type="button"
          >
            <span
              class="emotion-1"
            >
              <span
                aria-label="Close"
                class="sc-htpNat ebstqi"
                role="img"
              >
                <svg
                  focusable="false"
                  height="24"
                  role="presentation"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M15.185 7.4l-3.184 3.185-3.186-3.186a.507.507 0 0 0-.712.003l-.7.701a.496.496 0 0 0-.004.712l3.185 3.184L7.4 15.185a.507.507 0 0 0 .004.712l.7.7c.206.207.516.2.712.004l3.186-3.185 3.184 3.185a.507.507 0 0 0 .712-.004l.701-.7a.496.496 0 0 0 .003-.712l-3.186-3.186 3.186-3.184a.507.507 0 0 0-.003-.712l-.7-.7a.508.508 0 0 0-.36-.153.5.5 0 0 0-.353.15z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="emotion-14"
    >
      <div
        class="emotion-9"
      >
        <div
          class="emotion-8"
        >
          <div
            class="emotion-5"
          >
            <span
              class="sc-htpNat dcTkON"
              role="presentation"
            >
              <svg
                focusable="false"
                height="24"
                role="presentation"
                viewBox="0 0 24 24"
                width="24"
              >
                <path
                  d="M16.436 15.085l3.94 4.01a1 1 0 0 1-1.425 1.402l-3.938-4.006a7.5 7.5 0 1 1 1.423-1.406zM10.5 16a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
          <input
            autocomplete="off"
            class="emotion-6"
            name="help-search-input"
            placeholder="Search help articles"
            value=""
          />
          <div
            class="emotion-7"
          />
        </div>
      </div>
      <div
        class="emotion-13"
      >
        <div
          class="emotion-10"
          style="transition: opacity 440ms; opacity: 0; visibility: hidden;"
        />
        <div
          class="emotion-11"
          style="left: 100%; transition: left 300ms cubic-bezier(0.2, 0, 0, 1) 0s;"
        />
        <div
          class="emotion-12"
        >
          <div>
            Default content
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
